src/QsCompiler/QirGeneration/Subtransformations/ExpressionKindTransformation.cs

```diff
@@ -605,7 +684,29 @@ Value ArrayToString(ArrayValue array)
                 }
                 else if (ty.IsPauli)
                 {
-                    return SimpleToString(RuntimeLibrary.PauliToString);
+                    Value LoadPauli(QsPauli pauli)
```

Comment:
It seems like a rather complicated change to avoid calling `PauliToString`. Why the `Pauli` type needs to be handled this way?

Reply:
There are a couple of general considerations:
1) Generally, I think it is good to trim down what the runtime needs to implement to keep things simple for backends. 
2) If the runtime implements it, then the compiled code will break down into something similar, so there is no advantage or disadvantage really one way or the other after all is compiled.
3) The disadvantage of implementing it in the compiler is a little bit more code complexity compared to implementing it in the runtime (the code for us to maintain, not the code that a Q# project compiles into).

What the reason was for the immediate change is that with the current implementation of the runtime, there can be issues with bool and Pauli, since these don't match the bitsize used in the IR. This is only relevant in edge cases, and the proper solution is to revise the API to avoid such incompatibilities. We will do this as part of moving to a C-API. In the meantime, avoiding to use things we know can cause issues (for edge cases, most of the time the runtime functions work fine as is) is a mitigation.